### PR TITLE
[SYCL][NewOffloadingModel] Add sycl-dump-device-code command line option

### PR DESCRIPTION
**Pull request by maksimsab**
This option allows to save generated SPIRV files in the specified directory.

### CODE REVIEW

**Comment by maksimsab** [comment]
TODO: I need to add a change to Clang Driver to add usage of the added command line option.

File: clang/test/Driver/sycl-dump-device-code.cpp [hunk 1]
```diff
@@ -33,3 +33,36 @@
 
 // CHK-FSYCL-DUMP-DEVICE-CODE-CWD: llvm-foreach{{.*}} "--out-dir=.{{(/|\\\\)}}" "--" "{{.*}}llvm-spirv"
 // CHK-FSYCL-DUMP-DEVICE-CODE-WIN-CWD: llvm-foreach{{.*}} "--out-dir=.{{(/|\\\\)}}" "--" "{{.*}}llvm-spirv"
+
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-dump-device-code.cpp:36]: We should have a `sycl-dump-device-code-old-model.cpp` and `sycl-dump-device-code.cpp` testing (separate for old and new) to be consistent with other tests that have both functional behaviors.  It also makes it clear what is being tested based on the test name and allows for easier cleanup when we drop old model.

Comment (maksimsab) [reply clang/test/Driver/sycl-dump-device-code.cpp:36]: Makes sense.

Comment (maksimsab) [reply clang/test/Driver/sycl-dump-device-code.cpp]: Done.

**Review by mdtoguchi — APPROVED** [review]

File: clang/tools/clang-linker-wrapper/LinkerWrapperOpts.td [hunk 2]
```diff
@@ -220,3 +220,7 @@ Flags<[WrapperOnlyOption]>,  HelpText<"Link SYCL device code using thinLTO">;
 
 def sycl_embed_ir : Flag<["--", "-"], "sycl-embed-ir">,
 Flags<[WrapperOnlyOption]>,  HelpText<"Embed LLVM IR for runtime kernel fusion">;
+
+def sycl_dump_device_code_EQ : Joined<["--", "-"], "sycl-dump-device-code=">,
+  Flags<[WrapperOnlyOption]>,
+  HelpText<"Path to the folder where tool dumps SPIR-V device code. Other formats aren't being dumped.">;
```
Comment (sarnex) [inline clang/tools/clang-linker-wrapper/LinkerWrapperOpts.td:226]:
```suggestion
  HelpText<"Path to the folder where the tool dumps SPIR-V device code. Other formats aren't dumped.">;
```

**Review by sarnex — APPROVED** [review]
lgtm, just typo

File: clang/tools/clang-linker-wrapper/LinkerWrapperOpts.td (hunk 2 above)
Comment (maksimsab) [reply clang/tools/clang-linker-wrapper/LinkerWrapperOpts.td]: Fixed. Thanks.

**Comment by maksimsab** [comment]
@intel/llvm-gatekeepers Can we merge it?

**Pull request merged**